### ISSUE TITLE
False positive for SC2009 : should cope with the use of pgrep for collecting pids and then grepping process details

### ISSUE THREAD

**Issue by abpicoli1** [koalaman/shellcheck#2597]
#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
PID="$(ps -o pid,args -ww -p $(pgrep java) | grep com.ibm.is.log4j.SpaceChecker | grep normalcycle)"

```

#### Here's what shellcheck currently says: 
www.shellcheck output collected at Sept/28/2022
```
$ shellcheck myscript
 
[Line 2:](javascript:setPosition(2, 1))
PID="$(ps -o pid,args -ww -p $(pgrep java) | grep com.ibm.is.log4j.SpaceChecker | grep normalcycle)"
^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): PID appears unused. Verify use (or export if used externally).
       ^-- [SC2009](https://www.shellcheck.net/wiki/SC2009) (info): Consider using pgrep instead of grepping ps output.
                             ^-- [SC2046](https://www.shellcheck.net/wiki/SC2046) (warning): Quote this to prevent word splitting.

$
```

#### Here's what I wanted or expected to see:
SC2009 is not really an issue: the $(pgrep java) list the pids I want, making the main concern of SC2009 (to get the wrong process type) to be no longer an issue. In this example, I want to look for java processes that run a specific class and have a specific parameter.

**Comment by koalaman** [comment]
It's too bad `pgrep` doesn't have a `-p pidlist` flag. It would allow arbitrarily nested refinements like `pgrep -p "$(pgrep -p "$(pgrep java)" -f com.ibm.is.log4j.SpaceChecker)" -f normalcycle`. Instead, you no longer get a warning for `ps -p | grep`. Thanks!

**Comment by abpicoli1** [comment]
Well, if we want to discuss the operating system commands ... ideally we should have advanced queries inside a single command: what I want is something like "select processes.pid, processes.args where args[0] like '%/java' and args[x] like '%SpaceChecker%' and args[y] like 'normalcycle' and y > x". :) .

**Comment by Konfekt** [comment]
> It's too bad pgrep doesn't have a -p pidlist flag.

If I understand correctly, there's no grepping for PIDs with `pgrep`.
Therefore, the info could be more specific; rather than
```
Consider using pgrep instead of grepping ps output
```
something like
```
Consider using pgrep instead of grepping ps output to grep process names
```
